homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -2,53 +2,66 @@
 import logging
 
 from homeassistant.const import ATTR_ATTRIBUTION
-from homeassistant.core import callback
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.restore_state import RestoreEntity
-
-from .const import DATA_UPDATED, DOMAIN as SPEEDTESTDOTNET_DOMAIN, SENSOR_TYPES
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_BYTES_RECEIVED,
+    ATTR_BYTES_SENT,
+    ATTR_SERVER_COUNTRY,
+    ATTR_SERVER_ID,
+    ATTR_SERVER_NAME,
+    ATTRIBUTION,
+    DEFAULT_NAME,
+    DOMAIN,
+    ICON,
+    SENSOR_TYPES,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_BYTES_RECEIVED = "bytes_received"
-ATTR_BYTES_SENT = "bytes_sent"
-ATTR_SERVER_COUNTRY = "server_country"
-ATTR_SERVER_HOST = "server_host"
-ATTR_SERVER_ID = "server_id"
-ATTR_SERVER_LATENCY = "latency"
-ATTR_SERVER_NAME = "server_name"
 
-ATTRIBUTION = "Data retrieved from Speedtest.net by Ookla"
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Speedtestdotnet sensors."""
 
-ICON = "mdi:speedometer"
+    speedtestcoordinator = hass.data[DOMAIN]
 
+    entities = []
+    for sensor_type in SENSOR_TYPES:
+        entities.append(SpeedtestSensor(speedtestcoordinator, sensor_type))
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info):
-    """Set up the Speedtest.net sensor."""
-    data = hass.data[SPEEDTESTDOTNET_DOMAIN]
-    async_add_entities([SpeedtestSensor(data, sensor) for sensor in discovery_info])
+    async_add_entities(entities)
 
 
-class SpeedtestSensor(RestoreEntity):
+class SpeedtestSensor(Entity):
     """Implementation of a speedtest.net sensor."""
 
-    def __init__(self, speedtest_data, sensor_type):
+    def __init__(self, coordinator, sensor_type):
         """Initialize the sensor."""
         self._name = SENSOR_TYPES[sensor_type][0]
-        self.speedtest_client = speedtest_data
+        self.coordinator = coordinator
         self.type = sensor_type
         self._state = None
-        self._data = None
         self._unit_of_measurement = SENSOR_TYPES[self.type][1]
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return "{} {}".format("Speedtest", self._name)
+        return f"{DEFAULT_NAME} {self._name}"
+
+    @property
+    def unique_id(self):
+        """Return sensor unique_id."""
+        return self.type
 
     @property
     def state(self):
         """Return the state of the device."""
+        if self.type == "ping":
+            self._state = self.coordinator.data["ping"]
+        elif self.type == "download":
+            self._state = round(self.coordinator.data["download"] / 10 ** 6, 2)
+        elif self.type == "upload":
+            self._state = round(self.coordinator.data["upload"] / 10 ** 6, 2)
         return self._state
```

Comment:
State can just be a local variable since it's not used outside this property.
```suggestion
            state = self.coordinator.data["ping"]
        elif self.type == "download":
            state = round(self.coordinator.data["download"] / 10 ** 6, 2)
        elif self.type == "upload":
            state = round(self.coordinator.data["upload"] / 10 ** 6, 2)
        return state
```